server/controllers/refer/referralsController.test.ts

```diff
@@ -309,7 +309,7 @@ describe('ReferralsController', () => {
   describe('update', () => {
     describe('updating `oasysConfirmed`', () => {
       it('asks the service to update the field and redirects to the show action', async () => {
-        const referral = referralFactory.build({ oasysConfirmed: false })
+        const referral = referralFactory.build({ oasysConfirmed: false, reason: undefined })
```

Comment:
Does the line at the end of this block not need updating too? Seems to be passing 🤔 Could be a quirk of `undefined` - not defining it === `undefined`

```ts
expect(referralService.updateReferral).toHaveBeenCalledWith(token, referral.id, { oasysConfirmed: true })
```

Reply:
If it's `undefined`, it doesn't technically need to be specified, however if we think it will avoid confusion then happy to add `reason: undefined` and continue to do so in future.

Reply:
Do we _need_ to specify that `reason` is undefined here on line 312? Is it relevant to this section of the task list?

Reply:
Thought about it, probably nicer to remove the undefined and just update the test to check that `referralService.updateReferral` is called with the existing reason value.

Reply:
If we're [keeping a pre-filled](https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/193#discussion_r1324149829) `reason` as the factory default (I'm not sure what I think anymore 😅) then I think checking that it's called with the existing value makes sense here

Reply:
Related: https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/193/files#r1324300112